clientlibs/java/src/main/java/org/upgradeplatform/client/ExperimentClient.java

```diff
@@ -50,13 +50,12 @@
      *            Properties to permit users to control how the underlying JAX-RS
      *            client behaves. These are passed through to
      *            {@link javax.ws.rs.core.Configurable#property(String, Object)}. */
-	public ExperimentClient(String userId, String authToken, String baseUrl, Map<String, Object> properties) {
+	public ExperimentClient(String userId, String authToken, String sessionId, String baseUrl, Map<String, Object> properties) {
```

Comment:
In Javascript client we have kept sessionId optional so if it is not provided explicitly we can generate a random one.

Reply:
I added in a second constructor without the session id argument, making it optional.  If that one is used then we will generate a random one